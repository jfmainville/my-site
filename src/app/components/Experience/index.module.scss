@import "../../variables.module.scss";

.Experience {
  display: flex;
  flex-direction: column;
  background-color: $primary-color;
}

.ExperienceSection {
  display: grid;
  grid-template-columns: 1fr 7px 1fr;
}

.ExperienceSectionEntryLeft {
  display: flex;
  flex-direction: column;
  text-align: end;
  align-items: flex-end;
  color: $text-color;
  margin-right: 2rem;
}

.ExperienceSectionEntryRight {
  display: flex;
  flex-direction: column;
  text-align: start;
  align-items: flex-start;
  color: $text-color;
  margin-left: 2rem;
}

.ExperienceSectionEntryCircle {
  width: 1rem;
  height: 1rem;
  border-radius: 50%;
  background: $secondary-color;
}

.ExperienceSectionEntryLine {
  width: 0.1rem;
  height: 100%;
  background: $secondary-color;
}
.ExperienceSectionEntryLineRight {
  width: 0.1rem;
  height: 100%;
  margin-left: 7px;
  background: $secondary-color;
}

.ExperienceSectionEntryPeriod {
  text-decoration: underline;
}

.ExperienceSectionEntryRole {
  color: $secondary-color;
}

.TextHeading {
  color: $text-color;
  margin: 2rem;
}

.Link {
  font-size: 1.17em;
  color: $text-color;
  text-decoration: underline;
  padding: 0 1rem 0 1rem;
}
