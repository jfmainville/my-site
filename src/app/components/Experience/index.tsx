import styles from "./index.module.scss";
import bg from "../../../../public/images/roadmap.jpg";
import Image from "next/image";
import Link from "next/link";

const Experience = () => {
  return (
    <section
      id={"experience"}
      className={styles.Experience}
      style={{
        backgroundImage: `url(${bg.src})`,
        width: "100%",
        height: "auto",
      }}
    >
      <h1 className={styles.TextHeading}>Experience</h1>
      <div className={styles.ExperienceSection}>
        <div className={styles.ExperienceSectionEntryLeft}>
          <Image
            src="/images/cgi_logo.jpg"
            width={50}
            height={50}
            alt=""
            style={{
              height: "auto",
              margin: "0 0 1rem 0",
            }}
          />
          <p className={styles.ExperienceSectionEntryPeriod}>2013-2017</p>
          <p className={styles.ExperienceSectionEntryRole}>
            System Administrator
          </p>
          <p>
            As a System Administrator within the Microsoft Center of Expertise
            (MCoE) business unit, I specialized in managing Windows Server,
            SharePoint, SQL Server, and Microsoft Azure environments to ensure
            optimal performance and security
          </p>
        </div>
        <div className={styles.ExperienceSectionEntryCircle}></div>
        <div className={styles.ExperienceSectionEntryLine}></div>
        <div style={{ left: "10rem" }} />
        <div>
          <div className={styles.ExperienceSectionEntryCircle}></div>
          <div className={styles.ExperienceSectionEntryLineRight}></div>
        </div>
        <div className={styles.ExperienceSectionEntryRight}>
          <Image
            src="/images/cgi_logo.jpg"
            width={50}
            height={50}
            alt=""
            style={{
              height: "auto",
              margin: "0 0 1rem 0",
            }}
          />
          <p className={styles.ExperienceSectionEntryPeriod}>2013-2017</p>
          <p className={styles.ExperienceSectionEntryRole}>Security Analyst</p>
          <p>
            As a Security Analyst within the GTO business unit, I specialized in
            email security and monitoring to mitigate security risks
          </p>
        </div>
        <div className={styles.ExperienceSectionEntryLeft}>
          <Image
            src="/images/scfj_logo.jpg"
            width={50}
            height={50}
            alt=""
            style={{
              height: "auto",
              margin: "0 0 1rem 0",
            }}
          />
          <p className={styles.ExperienceSectionEntryPeriod}>2018-2019</p>
          <p className={styles.ExperienceSectionEntryRole}>
            Full-Stack Developer
          </p>
          <p>
            As a Full-Stack Developer, I developed an internal web application
            to streamline the time-tracking process for freelancers at SCFJ
          </p>
        </div>
        <div className={styles.ExperienceSectionEntryCircle}></div>
        <div className={styles.ExperienceSectionEntryLine}></div>
        <div style={{ left: "10rem" }} />
        <div>
          <div className={styles.ExperienceSectionEntryCircle}></div>
          <div className={styles.ExperienceSectionEntryLineRight}></div>
        </div>
        <div className={styles.ExperienceSectionEntryRight}>
          <Image
            src="/images/intact_logo.jpg"
            width={50}
            height={50}
            alt=""
            style={{
              height: "auto",
              margin: "0 0 1rem 0",
            }}
          />
          <p className={styles.ExperienceSectionEntryPeriod}>2019-2021</p>
          <p className={styles.ExperienceSectionEntryRole}>DevOps Specialist</p>
          <p>
            As a DevOps Specialist in the identity and access management (IAM)
            team, I contributed to enhancing the efficiency and security of the
            IAM infrastructure
          </p>
        </div>
        <div className={styles.ExperienceSectionEntryLeft}>
          <Image
            src="/images/intact_logo.jpg"
            width={50}
            height={50}
            alt=""
            style={{
              height: "auto",
              margin: "0 0 1rem 0",
            }}
          />
          <p className={styles.ExperienceSectionEntryPeriod}>2021-2022</p>
          <p className={styles.ExperienceSectionEntryRole}>
            Technical Product Owner
          </p>
          <p>
            As a Technical Product Owner in the security operations team, I lead
            my team in implementing and maintaining solutions to enhance
            Intact&apos;s security posture
          </p>
        </div>
        <div className={styles.ExperienceSectionEntryCircle}></div>
        <div className={styles.ExperienceSectionEntryLine}></div>
        <div style={{ left: "10rem" }} />
        <div>
          <div className={styles.ExperienceSectionEntryCircle}></div>
        </div>
        <div className={styles.ExperienceSectionEntryRight}>
          <Image
            src="/images/pixmob_logo.jpg"
            width={50}
            height={50}
            alt=""
            style={{
              height: "auto",
              margin: "0 0 1rem 0",
            }}
          />
          <p className={styles.ExperienceSectionEntryPeriod}>2022-Present</p>
          <p className={styles.ExperienceSectionEntryRole}>
            DevSecOps Specialist
          </p>
          <p>
            As the lead DevSecOps Specialist within the Software team, I
            designed and implemented the cloud infrastructure hosted on the
            Google Cloud Platform (GCP) and established robust security
            guidelines and infrastructure
          </p>
        </div>
      </div>
      <div
        style={{
          display: "flex",
          justifyContent: "center",
          margin: "3rem 0 3rem 0",
          textDecoration: "underline",
        }}
      >
        <Link className={styles.Link} href={"/resume"}>
          See my full resume
        </Link>
      </div>
    </section>
  );
};

export default Experience;
