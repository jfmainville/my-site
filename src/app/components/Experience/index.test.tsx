import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Experience from "./index";

describe("<Experience/> Test Suite", () => {
  it("should render the <Experience/> component unchanged", () => {
    const { container } = render(<Experience />);
    expect(container).toMatchSnapshot();
  });

  it("should show the correct amount of experiences in the <Experience /> component", () => {
    render(<Experience />);
    const roleSystemAdministrator = screen.getAllByText(
      "System Administrator",
      {
        exact: false,
      },
    );
    const roleSecurityAnalyst = screen.getAllByText("Security Analyst", {
      exact: false,
    });
    const roleFullStackDeveloper = screen.getAllByText("Full-Stack Developer", {
      exact: false,
    });
    const roleDevOpsSpecialist = screen.getAllByText("DevOps Specialist", {
      exact: false,
    });
    const roleTechnicalProductOwner = screen.getAllByText(
      "Technical Product Owner",
      {
        exact: false,
      },
    );
    const roleDevSecOpsSpecialist = screen.getAllByText(
      "DevSecOps Specialist",
      {
        exact: false,
      },
    );
    expect(roleSystemAdministrator).toHaveLength(2);
    expect(roleSecurityAnalyst).toHaveLength(2);
    expect(roleFullStackDeveloper).toHaveLength(2);
    expect(roleDevOpsSpecialist).toHaveLength(2);
    expect(roleTechnicalProductOwner).toHaveLength(2);
    expect(roleDevSecOpsSpecialist).toHaveLength(2);
  });

  it("should show the correct amount of company logos in the <Experience /> component", () => {
    render(<Experience />);
    const companyLogos = screen.getAllByRole("presentation");
    expect(companyLogos).toHaveLength(6);
  });
});
