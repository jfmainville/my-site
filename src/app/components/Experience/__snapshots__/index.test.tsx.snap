// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Experience/> Test Suite should render the <Experience/> component unchanged 1`] = `
<div>
  <section
    class="Experience"
    id="experience"
    style="background-image: url(/img.jpg); width: 100%; height: auto;"
  >
    <h1
      class="TextHeading"
    >
      Experience
    </h1>
    <div
      class="ExperienceSection"
    >
      <div
        class="ExperienceSectionEntryLeft"
      >
        <img
          alt=""
          data-nimg="1"
          decoding="async"
          height="50"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fcgi_logo.jpg&w=128&q=75"
          srcset="/_next/image?url=%2Fimages%2Fcgi_logo.jpg&w=64&q=75 1x, /_next/image?url=%2Fimages%2Fcgi_logo.jpg&w=128&q=75 2x"
          style="color: transparent; height: auto; margin: 0px 0px 1rem 0px;"
          width="50"
        />
        <p
          class="ExperienceSectionEntryPeriod"
        >
          2013-2017
        </p>
        <p
          class="ExperienceSectionEntryRole"
        >
          System Administrator
        </p>
        <p>
          As a System Administrator within the Microsoft Center of Expertise (MCoE) business unit, I specialized in managing Windows Server, SharePoint, SQL Server, and Microsoft Azure environments to ensure optimal performance and security
        </p>
      </div>
      <div
        class="ExperienceSectionEntryCircle"
      />
      <div
        class="ExperienceSectionEntryLine"
      />
      <div
        style="left: 10rem;"
      />
      <div>
        <div
          class="ExperienceSectionEntryCircle"
        />
        <div
          class="ExperienceSectionEntryLineRight"
        />
      </div>
      <div
        class="ExperienceSectionEntryRight"
      >
        <img
          alt=""
          data-nimg="1"
          decoding="async"
          height="50"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fcgi_logo.jpg&w=128&q=75"
          srcset="/_next/image?url=%2Fimages%2Fcgi_logo.jpg&w=64&q=75 1x, /_next/image?url=%2Fimages%2Fcgi_logo.jpg&w=128&q=75 2x"
          style="color: transparent; height: auto; margin: 0px 0px 1rem 0px;"
          width="50"
        />
        <p
          class="ExperienceSectionEntryPeriod"
        >
          2013-2017
        </p>
        <p
          class="ExperienceSectionEntryRole"
        >
          Security Analyst
        </p>
        <p>
          As a Security Analyst within the GTO business unit, I specialized in email security and monitoring to mitigate security risks
        </p>
      </div>
      <div
        class="ExperienceSectionEntryLeft"
      >
        <img
          alt=""
          data-nimg="1"
          decoding="async"
          height="50"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fscfj_logo.jpg&w=128&q=75"
          srcset="/_next/image?url=%2Fimages%2Fscfj_logo.jpg&w=64&q=75 1x, /_next/image?url=%2Fimages%2Fscfj_logo.jpg&w=128&q=75 2x"
          style="color: transparent; height: auto; margin: 0px 0px 1rem 0px;"
          width="50"
        />
        <p
          class="ExperienceSectionEntryPeriod"
        >
          2018-2019
        </p>
        <p
          class="ExperienceSectionEntryRole"
        >
          Full-Stack Developer
        </p>
        <p>
          As a Full-Stack Developer, I developed an internal web application to streamline the time-tracking process for freelancers at SCFJ
        </p>
      </div>
      <div
        class="ExperienceSectionEntryCircle"
      />
      <div
        class="ExperienceSectionEntryLine"
      />
      <div
        style="left: 10rem;"
      />
      <div>
        <div
          class="ExperienceSectionEntryCircle"
        />
        <div
          class="ExperienceSectionEntryLineRight"
        />
      </div>
      <div
        class="ExperienceSectionEntryRight"
      >
        <img
          alt=""
          data-nimg="1"
          decoding="async"
          height="50"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fintact_logo.jpg&w=128&q=75"
          srcset="/_next/image?url=%2Fimages%2Fintact_logo.jpg&w=64&q=75 1x, /_next/image?url=%2Fimages%2Fintact_logo.jpg&w=128&q=75 2x"
          style="color: transparent; height: auto; margin: 0px 0px 1rem 0px;"
          width="50"
        />
        <p
          class="ExperienceSectionEntryPeriod"
        >
          2019-2021
        </p>
        <p
          class="ExperienceSectionEntryRole"
        >
          DevOps Specialist
        </p>
        <p>
          As a DevOps Specialist in the identity and access management (IAM) team, I contributed to enhancing the efficiency and security of the IAM infrastructure
        </p>
      </div>
      <div
        class="ExperienceSectionEntryLeft"
      >
        <img
          alt=""
          data-nimg="1"
          decoding="async"
          height="50"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fintact_logo.jpg&w=128&q=75"
          srcset="/_next/image?url=%2Fimages%2Fintact_logo.jpg&w=64&q=75 1x, /_next/image?url=%2Fimages%2Fintact_logo.jpg&w=128&q=75 2x"
          style="color: transparent; height: auto; margin: 0px 0px 1rem 0px;"
          width="50"
        />
        <p
          class="ExperienceSectionEntryPeriod"
        >
          2021-2022
        </p>
        <p
          class="ExperienceSectionEntryRole"
        >
          Technical Product Owner
        </p>
        <p>
          As a Technical Product Owner in the security operations team, I lead my team in implementing and maintaining solutions to enhance Intact's security posture
        </p>
      </div>
      <div
        class="ExperienceSectionEntryCircle"
      />
      <div
        class="ExperienceSectionEntryLine"
      />
      <div
        style="left: 10rem;"
      />
      <div>
        <div
          class="ExperienceSectionEntryCircle"
        />
      </div>
      <div
        class="ExperienceSectionEntryRight"
      >
        <img
          alt=""
          data-nimg="1"
          decoding="async"
          height="50"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fpixmob_logo.jpg&w=128&q=75"
          srcset="/_next/image?url=%2Fimages%2Fpixmob_logo.jpg&w=64&q=75 1x, /_next/image?url=%2Fimages%2Fpixmob_logo.jpg&w=128&q=75 2x"
          style="color: transparent; height: auto; margin: 0px 0px 1rem 0px;"
          width="50"
        />
        <p
          class="ExperienceSectionEntryPeriod"
        >
          2022-Present
        </p>
        <p
          class="ExperienceSectionEntryRole"
        >
          DevSecOps Specialist
        </p>
        <p>
          As the lead DevSecOps Specialist within the Software team, I designed and implemented the cloud infrastructure hosted on the Google Cloud Platform (GCP) and established robust security guidelines and infrastructure
        </p>
      </div>
    </div>
    <div
      style="display: flex; justify-content: center; margin: 3rem 0px 3rem 0px; text-decoration: underline;"
    >
      <a
        class="Link"
        href="/resume"
      >
        See my full resume
      </a>
    </div>
  </section>
</div>
`;
